### PR TITLE
Turn off `strings_as_factors_linter()`

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Since we rely on >= R 4.0 now.

### CODE REVIEW

**Comment by MichaelChirico** [comment]
Confirming this is everything re-using the _ad hoc_ linter in https://github.com/r-lib/lintr/pull/2605:

```r
l <- make_linter_from_xpath(lint_message = "", "
  //SYMBOL_SUB[text() = 'stringsAsFactors']
    /parent::expr[not(
      expr/SYMBOL_FUNCTION_CALL[text() = 'expand.grid']
      or ./expr/expr/SYMBOL_FUNCTION_CALL[text() = 'table']
    )]
  ")
lint_package(linters = l())
```

File: R/ui-caching.R [hunk 1]
```diff
@@ -104,8 +104,7 @@ cache_info <- function(cache_name = NULL, format = "both") {
     last_modified = suppressWarnings(max(file_info$mtime)),
     created = file.info(path_cache)$ctime,
     location = path_cache,
-    activated = cache_is_activated(cache_name),
-    stringsAsFactors = FALSE
```
Comment (MichaelChirico) [inline R/ui-caching.R:108]: IINM this is not WAI anyway?

```
styler:::styler_df(a = 1:10, b = 11:20, stringsAsFactors=FALSE)
    a  b stringsAsFactors
1   1 11            FALSE
2   2 12            FALSE
3   3 13            FALSE
4   4 14            FALSE
5   5 15            FALSE
6   6 16            FALSE
7   7 17            FALSE
8   8 18            FALSE
9   9 19            FALSE
10 10 20            FALSE
```

Comment (IndrajeetPatil) [reply R/ui-caching.R:108]: That's true!

**Review by MichaelChirico — APPROVED** [review]
Feel free to merge, the part about `styler_df()` can be handled here or in another PR, up to you :)

**Review by lorenzwalthert — APPROVED** [review]

**Pull request merged**